Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +60,135 @@ class TransactionHistoryViewController: UIViewController {
     
     override func viewDidAppear(_ animated: Bool) {
         if !AppDelegate.walletLoader.isSynced {
-            print(" wallet not synced on history")
+            print("wallet not synced on history")
             return
         }
-        self.visible = true
-        if (self.FromMenu){
-            refreshControl.showLoader(in: self.tableView)
-            loadTransactions()
-            FromMenu = true
+        self.refreshControl.showLoader(in: self.tableView)
+        loadAllTransactions()
+    }
+    
+    func initFilterBtn() {
+        self.btnFilter.initMenu(self.filterOptions) { [weak self] index, value in
+            self?.applyTxFilter(currentFilter: self!.filters[index])
+        }
+    }
+    
+    func loadAllTransactions() {
+        self.allTransactions.removeAll()
+        var error: NSError?
+        
+        let allTransactions = AppDelegate.walletLoader.wallet?.getTransactions(0, txFilter: DcrlibwalletTxFilterAll, error: &error)
+        if error != nil {
+            print(error!.localizedDescription)
+        }
+        
+        if allTransactions == nil || allTransactions!.count == 0 {
+            self.showNoTransactions()
+            return
+        }
+        
+        self.allTransactions = try! JSONDecoder().decode([Transaction].self, from: allTransactions!.data(using: .utf8)!)
+        
+        self.tableView.backgroundView = nil
+        self.tableView.separatorStyle = .singleLine
+        self.refreshControl.endRefreshing()
+        self.updateFilterDropdownItems()
+        self.tableView.reloadData()
+    }
+    
+    func applyTxFilter(currentFilter: Int32) {
+        refreshControl.showLoader(in: self.tableView)
+        filteredTransactions.removeAll()
+        
+        self.filterActive = currentFilter != DcrlibwalletTxFilterAll
+        
+        switch currentFilter {
+        case DcrlibwalletTxDirectionSent:
+            self.filteredTransactions = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionSent && $0.Type == DcrlibwalletTxTypeRegular}
+            self.btnFilter.setTitle(LocalizedStrings.sent.appending("(\(self.filteredTransactions.count))"), for: .normal)
```

Comment:
Are each of these `btnFilter.setTitle` calls necessary? When a user interacts with the dropdown and selects an item, the underlying button would/should automatically have its title set to the selected item text.

Reply:
haven't figured out a way to deal with the string differences yet.

Reply:
Not sure I understand what you mean.

Reply:
each button instance contains a different text that is being set from a localized string. i have to call setTitle each time because of this.